### ISSUE TITLE
Credentials expiration date is not set since the expires_in value is an Int64

### ISSUE THREAD

**Issue by seanmcneil** [auth0/Auth0.swift#439]
### Describe the problem

The expiration date is not being set for the Credentials object upon initialization since the provided data type is not handled.

![credentials](https://user-images.githubusercontent.com/6424532/105079879-e6b79900-5a44-11eb-95ba-57a1beda43a8.png)

### What was the expected behavior?

I expect the Credentials object to set a valid expiration date.

### Reproduction

> Detail the steps taken to reproduce this error, and whether this issue can be reproduced consistently or if it is intermittent.

- Step 1 Authenticate
- Step 2 Examine initialization of Credentials object in 
- `convenience required public init(json: [String: Any]) { }`

**Can the behavior be reproduced using the [Auth0.swift sample app](https://github.com/auth0-samples/auth0-ios-swift-sample)?**

### Environment

> Please provide the following:

- 1.30.1
- 14.2
- 12.2

I have a PR ready that resolves this and will publish shortly.

**Comment by seanmcneil** [comment]
I've also found that there appears to be another issue going on with this. 

While I can test the code snippet independently in a playground and it works, or the associated tests all pass, it fails when I pull the pod into my project. I tracked the issue down to an unusual error with instantiating a `Date` object, thus preventing it from modifying the date.

 It turns out, that possibly due to the @Objc compatibility, the class cannot use `Date`, but must instead use `NSDate`. I found that by changing the relevant data types to match `NSDate`, that only then would it successfully produce an expiration date. This is not connected to the issue I saw or the work I did. The class simply cannot access `Date`.

With that said, the pod in its current form still fails for me. I could convert my work to use `NSDate`, but am not sure if there is something I may be missing here?

![Screen Shot 2021-01-19 at 2 06 05 PM](https://user-images.githubusercontent.com/6424532/105098804-7c602200-5a5f-11eb-8c9c-eb01b96c9dae.png)
